gralloc_gbm.cpp

```diff
@@ -482,3 +482,50 @@ int gralloc_gbm_bo_unlock(buffer_handle_t handle)
 
 	return 0;
 }
+
+#define GRALLOC_ALIGN(value, base) (((value) + ((base)-1)) & ~((base)-1))
+
+int gralloc_gbm_bo_lock_ycbcr(buffer_handle_t handle,
+		int usage, int x, int y, int w, int h,
+		struct android_ycbcr *ycbcr)
+{
+	struct gralloc_handle_t *hnd = gralloc_handle(handle);
+	int ystride, cstride;
+	void *addr;
+	int err;
+
+	ALOGD("handle %p, hnd %p, usage 0x%x", handle, hnd, usage);
+
+	err = gralloc_gbm_bo_lock(handle, usage, x, y, w, h, &addr);
+	if (err)
+		return err;
+
+	memset(ycbcr->reserved, 0, sizeof(ycbcr->reserved));
+
+	switch (hnd->format) {
+	case HAL_PIXEL_FORMAT_YCrCb_420_SP:
+		ystride = cstride = GRALLOC_ALIGN(hnd->width, 16);
+		ycbcr->y = addr;
+		ycbcr->cr = (unsigned char *)addr + ystride * hnd->height;
+		ycbcr->cb = (unsigned char *)addr + ystride * hnd->height + 1;
+		ycbcr->ystride = ystride;
+		ycbcr->cstride = cstride;
+		ycbcr->chroma_step = 2;
+		break;
+	case HAL_PIXEL_FORMAT_YV12:
+		ystride = hnd->width;
+		cstride = GRALLOC_ALIGN(ystride / 2, 16);
```

Comment:
@pundiramit , 
Hello,
I just want to ask why does 16 byte alignment has chosen.
I had issues with youtube playback on webview browser until I raised it up to 128 bytes.
(On lima GPU driver)

Reply:
Ah, It also turns out that `mesa3d` uses addresses as offsets.
I will post a fix into mesa3d today.

Reply:
https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/6254

Reply:
I just saw the comment in mesa3d:
```
/* When lock_ycbcr's usage argument contains no SW_READ/WRITE flags
    * it will return the .y/.cb/.cr pointers based on a NULL pointer,
    * so they can be interpreted as offsets. */
```
For some reason it does not happen.

Reply:
https://cs.android.com/android/platform/superproject/+/master:hardware/libhardware/include/hardware/gralloc.h;l=204;drc=5543b1eb045a4946449982de99474a441fdc1e9d